app/assets/javascripts/map.js

```diff
@@ -125,25 +207,210 @@ class Map {
         position: results[0].geometry.location
       });
       this.geocodedAddress = results[0].formatted_address;
-      this.onGeocodingResponse(this.geocodedAddress);
+      this.updateAddressOnSummary(this.geocodedAddress);
     } else {
       alert('Geocode was not successful for the following reason: ' + status);
     }
   }
 
+  updateAddressOnSummary(geocodedAddress) {
+    if (geocodedAddress) {
+      this.geocodedAddress = geocodedAddress;
+      const primaryAddressNode = document.getElementById('primaryAddress');
+      const secondaryAddressNode = document.getElementById('secondaryAddress');
+      while (primaryAddressNode.firstChild) { primaryAddressNode.removeChild(primaryAddressNode.firstChild) }
+      while (secondaryAddressNode.firstChild) { secondaryAddressNode.removeChild(secondaryAddressNode.firstChild) }
```

Comment:
I would also add this to the `Dom` class, with something like `empty(node)` or something like that.